migrations/src/main/scala/za/co/absa/enceladus/migrations/continuous/migrate01/EntityRepository.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.migrations.continuous.migrate01
+
+import org.mongodb.scala.MongoDatabase
+import org.mongodb.scala.bson.{BsonDocument, ObjectId}
+import org.mongodb.scala.model.Filters
+import org.mongodb.scala.model.Filters.{and, equal, or}
+import org.mongodb.scala.model.Sorts._
+import za.co.absa.atum.model.RunState
+import za.co.absa.enceladus.migrations.framework.ObjectIdTools
+
+/**
+  * The class contains a set of MongoDB versioned collection manipulation routines needed for continuous migration.
+  */
+final class EntityRepository(db: MongoDatabase, collectionName: String) {
+
+  import za.co.absa.enceladus.migrations.framework.dao.ScalaMongoImplicits._
+
+  /**
+    * Returns iterator to a sorted collection of documents. Documents are sorted by name and version.
+    *
+    * @return An iterator to a JSON representation of the documents.
+    */
+  def getSortedDocuments: Iterator[String] = {
+    db.getCollection(collectionName)
+      .find()
+      .sort(ascending("name", "version"))
+      .execute()
+      .toIterator
+      .map(_.toJson)
+  }
+
+  /**
+    * Returns iterator to a sorted collection of runs. Documents are sorted by datasetName, datasetVersion and runId.
+    *
+    * @return An iterator to a JSON representation of the documents.
+    */
+  def getSortedRuns: Iterator[String] = {
+    db.getCollection(collectionName)
+      .find(
+        or(
+          equal("runStatus.status", RunState.allSucceeded.toString),
+          equal("runStatus.status", RunState.failed.toString)
+        )
+      )
+      .sort(ascending("dataset", "datasetVersion", "runId"))
+      .execute()
+      .toIterator
+      .map(_.toJson)
+  }
+
+  /**
+    * Returns true if a document with the particular Object Id exists in the collection.
+    *
+    * @param objectId An object Id of an entity.
+    * @return true if such a document exists, false otherwise.
+    */
+  def doesDocumentExist(objectId: String): Boolean = {
+    val id = ObjectIdTools.extractId(objectId)
+    val documents = db.getCollection(collectionName)
+      .find(Filters.equal("_id", new ObjectId(id)))
+      .execute()
+    documents.nonEmpty
+  }
+
+  /**
+    * Returns true if a document with the particular name and version exists in the collection.
+    *
+    * @param name    A name of an entity.
+    * @param version A version of the entity.
```

Comment:
```suggestion
    * @param version A version of an entity.
```
It's odd to see `the` there, especially after the previous line is analogous but has `an entity` in it. You have it like this in all following cases as well.